src/components/date-range-picker/date-range.service.ts

```diff
@@ -45,6 +45,12 @@ export class DateRangeService {
     /** Indicate if we are currently changing the time */
     isChangingTime: boolean = false;
 
+    /** Store the current start time */
+    startTime: DateRangeTime = { hours: 0, minutes: 0, seconds: 0 };
```

Comment:
Could this not be part of `start`/`end` above?

Reply:
I had to keep them separate because if I select a start date, but do not have an end date (or vice versa), then when I hover over a date in the end picker it shows the duration from the start date to the hover date, so in this instance setting it directly on the `end` property would cause an end date to actually be chosen, rather than just showing a preview of what the duration would be, if that makes sense?

Reply:
Understood! Thanks for the clarification.